recipes/ogdf/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def validate(self):
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("coin-clp/1.17.7")
+        self.requires("pugixml/1.13")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.variables["COIN_SOLVER"] = "CLP"
+        tc.variables["COIN_SOLVER_IS_EXTERNAL"] = 0
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    def _patch_sources(self):
+        # delete code from other cci packages
+        rmdir(self, join(self.source_folder, "src", "coin"))
+        rmdir(self, join(self.source_folder, "include", "coin"))
+        rmdir(self, join(self.source_folder, "include", "ogdf", "lib", "backward"))
+        rmdir(self, join(self.source_folder, "src", "ogdf", "lib", "pugixml"))
+        rmdir(self, join(self.source_folder, "include", "ogdf", "lib", "pugixml"))
+        # use cci packages where available
+        replace_in_file(self, join(self.source_folder, "CMakeLists.txt"), "include(coin)", "find_package(coin-clp REQUIRED CONFIG)\nfind_package(pugixml REQUIRED CONFIG)")
+        replace_in_file(self, join(self.source_folder, "cmake", "ogdf.cmake"), "target_link_libraries(OGDF PUBLIC COIN)", "target_link_libraries(OGDF PUBLIC coin-clp::coin-clp pugixml::pugixml)")
+        # replace pugixml copy in repo by conan dependency
+        for dir_name, file_name in [("include", "GexfParser.h"),
+                                    ("include", "GraphMLParser.h"),
+                                    ("include", "SvgPrinter.h"),
+                                    ("include", "TsplibXmlParser.h"),
+                                    ("src", "GraphIO_graphml.cpp"),
+                                    ("src", "GraphIO_gexf.cpp")]:
+            replace_in_file(self, join(self.source_folder, dir_name, "ogdf", "fileformats", file_name), "ogdf/lib/pugixml/pugixml.h", "pugixml.hpp")
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build(target="OGDF")
+
+    def package(self):
+        copy(self, pattern="LICENSE.txt", src=self.source_folder, dst=join(self.package_folder, "licenses"))
+        copy(self, pattern="*.h", src=join(self.source_folder, "include"), dst=join(self.package_folder, "include"))
```

Comment:
Why do you use `cmake.install()` instead? I see it supported by the upstream: https://github.com/ogdf/ogdf/blob/master/cmake/ogdf.cmake#L202

Reply:
I will test that after #15836 is merged. This blocks the build currently

Reply:
https://github.com/conan-io/conan-center-index/pull/15836 is ready for review, it passes v1 & v2 pipeline now.

Reply:
I gave it a go. It would work with cmake install, but that would add some test instances, some code examples, and copy the CMake config the OGDF project itself creates (I am not sure whether that is compatible with Conan's CMake):

- copies all files from https://github.com/ogdf/ogdf/tree/master/doc/examples
- creates the cmake files `OgdfTargets.cmake`, `OgdfTargets-release.cmake` and `ogdf-config.cmake` defined by https://github.com/ogdf/ogdf/blob/master/cmake/ogdf.cmake#L201

What should I do? Copy files by hand as in the recipe in its current form. Or use cmake install as is. Or use cmake install and then delete files afterwards?

Reply:
patches introduce maintenance burden, they have to be constantly rebased for each new releases.
I would only use them as a last resort, when all the other ways failed to work already.
patches make PRs harder to review in general, as they introduce an additional risk of diverging from upstream's behavior (you might remove some CMake targets from install, but they may have affected on another CMake logic by constructions like `if (TARGET xxx)`).

Reply:
Thats for sharing your thoughts however the "cost" is far smaller then the risk of mistakes and quality.

Patches should be reported upstream when appropriate and tracked accordingly with the metadata fields so that consumers can evaluate their efficacy :)

The return for using patch files is why they should be preferred in all cases.

Reply:
Very well put! Theres valid cases for both - but I will maintain there's a preference for actual patch files :)